### PR TITLE
Reporter refactor

### PR DESCRIPTION
**Pull request by netmajor**

### CODE REVIEW

File: lib/reporter.js [hunk 1]
```diff
@@ -25,145 +25,190 @@ class AllureReporter extends events.EventEmitter {
...
+        this.on('allure:feature', this.allureFeature)
+        this.on('allure:addEnvironment', this.allureAddEnvironment)
+        this.on('allure:attachment', this.allureAttachment)
+    }
 
-        this.on('suite:end', (suite) => {
-            this.getAllure(suite.cid).endSuite()
-        })
+    bindEventFunctions () {
+        this.hookStart = this.hookStart.bind(this)
```
Comment (christian-bromann) [inline lib/reporter.js]: Let's do

```js
this.hookStart = ::this.hookStart
// ...
```

Comment (netmajor) [reply lib/reporter.js]: This us still poc, so I am using older but stable syntax.

I am trying to revise master to my branch but Ci build failed, don't know why :/

Comment (netmajor) [reply lib/reporter.js]: https://github.com/netmajor/wdio-allure-reporter/pull/4

https://travis-ci.org/netmajor/wdio-allure-reporter/builds/330083718?utm_source=github_status&utm_medium=notification

Comment (netmajor) [reply lib/reporter.js]: When I tried to merge from master to my branch i get : There isn’t anything to compare.

Comment (christian-bromann) [reply lib/reporter.js]: just switch to master, pull latest code, create a branch, apply changes and create a new PR .. I guess this is the easiest

Comment (netmajor) [reply lib/reporter.js]: Ok. I will try it tomorrow.

**Pull request closed without merging**